lib/rest-router.js

```diff
@@ -0,0 +1,59 @@
+
+var url = require('url');
+var http = require('http');
+var Trie = require('./trie');
+
+var HTTP_METHODS = require('methods');
+
+var RestRouter = module.exports = function(options) {
+  var opts = options || {};
+  var router = function(req, res, next) {
+    router.handle(req, res, next);
+  };
+
+  router.__proto__ = RestRouter;
+  router._trie = new Trie();
+  router.options = opts;
+
+  return router;
+};
+
+// express style API for registering route handlers: ex. route[method](path, handler)
+// add verb:handler for path to trie DS
+HTTP_METHODS.concat('all').forEach(function(method) {
+  RestRouter[method] = function(route, handler) {
+    route.fullPath = normalizePath(route.fullPath, this.opts);
+    var trie = this._trie;
+    trie = trie.add(route, handler);
+    return this;
+  };
+});
+
+// handle request, match path and if found, invoke handler method
+RestRouter.handle = function(req, res, next) {
+  var trie = this._trie;
+  var verb = req.method.toLowerCase();
+  var path = normalizePath(req.url, this.options);
+  var methods = trie.find(path);
+
+  if (methods && verb in methods) {
+    methods[verb](req, res, next);
+  } else {
+    next();
+  }
+};
+
+// For now: exclude paths with parameters like :id
+// and handler functions with err obj i.e. (err, req, res, next)
```

Comment:
You commented-out  `if (router.canRegister`)` , therefore no paths/handlers are excluded now AFAICT. I find this comment confusing, because it describes situation before you made your changes and seems to be in contradiction with the current version of the code.